### DISCUSSION TITLE
Blog post titles to be links

### DISCUSSION THREAD

**Discussion by hellt** [squidfunk/mkdocs-material#4395] [Ideas]
I take it that the doc-site ancestry made headings non clickable, but with a blog dressing it makes sense to make titles to be clickable.
Right now if anyone wants to drill down to the post, they have to aim for "Continue reading" link if the extract is available. I wonder if you think that making blog titles act like links make sense

**Comment by squidfunk** [comment]
In general, yes. Some users might want to have it, other users not, though. Additionally, we would need to jump through some serious hoops to properly support this for the blog plugin. The reason is that we render post excerpts from the articles themselves, and authors are completely free in what they include in the excerpt – they don't even need to set a `h1`, or they can insert images or other content before the `h1`. Thus, what we would have to do is to parse the content and wrap it with a link, if we would support this as part of the blog plugin. I'm unsure what we should do here, but I take this with me into the iteration of the plugin and see if we can offer a comfortable way of doing this.

**Reply by hellt** [reply]
maybe it is possible to override the permalink link to point to the actual post page? Currently permalink links to the heading section on the index page, which doesn't make a lot of sense blog-wise.
I can see how overriding permalink might be seen as a bad practice, but just wondering if this can be done.

**Reply by squidfunk** [reply]
Permalinks should be switched off when generating the blog index pages. We might somehow (mis-)use the `toc` extension that provides permalink support, but it's definitely non-trivial.

**Reply by hellt** [reply]
Fixed in #4545
